crypto/x509/x509_vfy.c

```diff
@@ -3004,7 +2958,8 @@ static int build_chain(X509_STORE_CTX *ctx)
      * typically the content of the peer's certificate message) so can make
      * multiple passes over it, while free to remove elements as we go.
      */
-    if (ctx->untrusted && (sktmp = sk_X509_dup(ctx->untrusted)) == NULL) {
+    if (ctx->untrusted != NULL
+            && (sk_untrusted = sk_X509_dup(ctx->untrusted)) == NULL) {
```

Comment:
At first I found that weird, too, but there is a special OpenSSL coding guideline about it.
It does make sense because otherwise the second line of the condition is indented exactly like the following statement that depends on it, making a bit hard to read where the condition ends and the subsequent statement/block starts.

Reply:
I just looked it up: that exception rule is not yet officially part of the coding guidelines but is likely going to be, see #10725.
At least, many places in the code are indented that way, and for some earlier PRs I was asked to use the extra indentation before a leading `&&` (or `||`, respectively).

Reply:
OK, although I do not think this rule will ever be made mandatory.

Reply:
That's what I meant: 
an explicitly allowed possibility (to deviate from normal indentation, for readability).